src/views/TechnicalAssistance/AdviceLetterForm/NextSteps.tsx

```diff
@@ -61,55 +61,59 @@ const NextSteps = ({
     }
   });
 
+  const handleApolloError = useCallback(
+    (err: any) => {
+      if (err instanceof ApolloError) {
+        setFormAlert({
+          type: 'error',
+          message: t('adviceLetterForm.error', {
+            action: 'saving',
+            type: 'advice letter'
+          })
+        });
+      }
+    },
+    [setFormAlert, t]
+  );
+
   /** Submit next steps fields and update advice letter */
-  const submit: StepSubmit = useCallback(
-    callback => {
-      /** Submits form and updates advice letter */
-      const submitForm = handleSubmit(
+  const submit = useCallback<StepSubmit>(
+    callback =>
+      handleSubmit(
         async formData => {
-          if (isDirty) {
-            // UpdateTrbAdviceLetter mutation
-            await update({
-              variables: {
-                input: {
-                  trbRequestId,
-                  ...formData,
-                  // If isFollowUpRecommended is set to false, clear followupPoint value
-                  followupPoint: formData.isFollowupRecommended
-                    ? formData.followupPoint
-                    : null
+          try {
+            if (isDirty) {
+              // UpdateTrbAdviceLetter mutation
+              await update({
+                variables: {
+                  input: {
+                    trbRequestId,
+                    ...formData,
+                    // If isFollowUpRecommended is set to false, clear followupPoint value
+                    followupPoint: formData.isFollowupRecommended
+                      ? formData.followupPoint
+                      : null
+                  }
                 }
-              }
-            });
+              });
+            }
+            setFormAlert(null);
+            callback?.();
+          } catch (e) {
+            handleApolloError(e);
           }
         },
         // Throw error to cause promise to fail
-        () => {
-          throw new Error('Invalid field submission');
-        }
-      );
-
-      return submitForm().then(
-        // If successful, set error to null and execute callback
-        () => {
-          setFormAlert(null);
-          callback?.();
-        },
-        // If apollo error, set form alert error message
-        e => {
-          if (e instanceof ApolloError) {
-            setFormAlert({
-              type: 'error',
-              message: t('adviceLetterForm.error', {
-                action: 'saving',
-                type: 'advice letter'
-              })
-            });
-          }
-        }
-      );
-    },
-    [handleSubmit, isDirty, trbRequestId, update, setFormAlert, t]
+        e => handleApolloError(e)
+      )(),
```

Comment:
IIRC you said this was the handler for validation error.  I believe validation error handling is inherent.  This handler checks for instances of Apollo error to render anything as well.  Removing this seems to have no effect on error nor validation handling.  When I log the error on validation error, I also get nothing.  I don't think it hurts to have it in there, but I'm also not certain it's doing anything with the current setup.

Reply:
Good catch! You're right, handling the error there isn't necessary anymore. I updated the code to remove it.